File: Python/board/initialize_board.py
# -*- coding: utf-8 -*-
"""
Created on Mon Oct  3 14:23:58 2022

@author: Swen
"""
import numpy as np

from common.constants import VALID_SQUARES
from common.names import OUTER, WHITE, BLACK, EMPTY

def initialize_board():
        """
            Create a new board with the initial black and white possitions.
            Returns a list corresponding to each element.
        """
        board = [OUTER] * 100
        for i in VALID_SQUARES:
            board[i] = EMPTY
        # the middle four squares should hold the initial piece positions.
        board[44], board[45] = WHITE, BLACK
        board[54], board[55] = BLACK, WHITE
        return np.array(board)

File: Cpp/board/initialize_board.cc
#include "board.ih"

void initialize_board(char* board)
{
    for (int idx = 0; idx != 100; ++idx)
    {
        int col = idx % 10;                                 // get col index
        int row = idx / 10;                                 // row idx

        bool col_between = col > 0 && col < 9;              // borders
        bool row_between = row > 0 && row < 9;  

        // set empty or border
        board[idx] = (col_between && row_between)?EMPTY:OUTER;  
    }

    // fill special cases
    board[44] = WHITE; 
    board[55] = WHITE;
    board[45] = BLACK;
    board[54] = BLACK;
}